src/astgen.zig

```diff
@@ -267,6 +267,10 @@ pub fn expr(mod: *Module, scope: *Scope, rl: ResultLoc, node: ast.Node.Index) In
     const node_tags = tree.nodes.items(.tag);
     const token_starts = tree.tokens.items(.start);
 
+    // TODO since some of the tags don't need a gen_zir (the const ones)
+    // should it be better to do this in the individual switch cases?
```

Comment:
Not necessary. The optimizer is fully capable of moving this to inside the branches. It's also supposed to be a cheap function to call; I doubt it will be noticeable in terms of perf. If it is, what we really need to do is introduce a new type that we pass around in astgen.zig that makes the GenZir which is in scope always be directly available rather than accessible via an indirect branch from `*Scope`.